src/llama.cpp

```diff
@@ -6473,6 +6475,11 @@ static void llm_load_vocab(
             } else if (
                     tokenizer_pre == "falcon") {
                 vocab.type_pre = LLAMA_VOCAB_PRE_TYPE_FALCON;
+            } else if (
+                    tokenizer_pre == "falcon3") {
+                vocab.type_pre = LLAMA_VOCAB_PRE_TYPE_LLAMA3;
+                vocab.tokenizer_ignore_merges = true;
+                vocab.tokenizer_add_bos = true;
```

Comment:
Let's simplify this by moving the check to the `LLAMA3` branch above:

```c++
                    tokenizer_pre == "llama3"    ||
                    tokenizer_pre == "llama-v3"  ||
                    tokenizer_pre == "llama-bpe" ||
                    tokenizer_pre == "falcon3") {
```

Reply:
Done.